diff --git a/org.kde.tokodon.desktop b/org.kde.tokodon.desktop
index 49d98102..01c2f2eb 100644
--- a/org.kde.tokodon.desktop
+++ b/org.kde.tokodon.desktop
@@ -110,7 +110,7 @@ Comment[uk]=Клієнт мережі Mastodon
 Comment[x-test]=xxClient for the Mastodonxx
 Comment[zh_CN]=Mastodon 客户端
 Comment[zh_TW]=Mastodon 用戶端
-Exec=tokodon %u
+Exec=env KDE_FULL_SESSION=1 KDE_SESSION_VERSION=6 qt-runner tokodon %u
 Terminal=false
 Icon=org.kde.tokodon
 Type=Application
@@ -118,3 +118,8 @@ Categories=Network;InstantMessaging;X-SocialMedia
 X-KDE-FormFactor=desktop;tablet;handset;
 SingleMainWindow=true
 MimeType=x-scheme-handler/web+ap;x-scheme-handler/tokodon;
+X-Nemo-Single-Instance=no
+X-Nemo-Application-Type=no-invoker
+
+[X-Sailjail]
+Sandboxing=Disabled
